src/api/annotation/AnnotationAPI.js

```diff
@@ -0,0 +1,319 @@
+
+        return existingAnnotation;
+    }
+
+    async removeNotebookAnnotationTag(entryId, targetDomainObject, tagToRemove) {
+        let existingAnnotation = await this.getNotebookAnnotation(entryId, targetDomainObject);
+
+        if (existingAnnotation && existingAnnotation.tags.includes(tagToRemove)) {
+            const cleanedArray = existingAnnotation.tags.filter(extantTag => extantTag !== tagToRemove);
+            this.openmct.objects.mutate(existingAnnotation, 'tags', cleanedArray);
+        } else {
+            throw new Error(`Asked to remove notebook tag (${tagToRemove}) that doesn't exist for ${entryId}`);
+        }
+    }
+
+    async removeNotebookAnnotation(entryId, targetDomainObject) {
+        // just removes tags on the annotation as we can't really delete objects
+        const notebookAnnotationToRemove = await this.getNotebookAnnotation(entryId, targetDomainObject);
+        if (notebookAnnotationToRemove && notebookAnnotationToRemove.tags) {
+            this.openmct.objects.mutate(notebookAnnotationToRemove, 'tags', []);
+        }
+    }
+
+    getMatchingTags(query) {
+        if (!query) {
+            return [];
+        }
+
+        const allTags = availableTags.tags;
+        const matchingTags = Object.keys(allTags).filter(tagKey => {
+            if (allTags[tagKey] && allTags[tagKey].label) {
+                return allTags[tagKey].label.toLowerCase().includes(query.toLowerCase());
+            }
+
+            return false;
+        });
+
+        return matchingTags;
+    }
+
+    addTagMetaInformationToResults(results, matchingTagKeys) {
+        const tagsAddedToResults = results.map(result => {
+            const fullTagModels = result.tags.map(tagKey => {
+                const tagModel = availableTags.tags[tagKey];
+                tagModel.tagID = tagKey;
+
+                return tagModel;
+            });
+
+            return {
+                fullTagModels,
+                matchingTagKeys,
+                ...result
+            };
+        });
+
+        return tagsAddedToResults;
+    }
+
+    async addTargetModelsToResults(results) {
+        const modelAddedToResults = await Promise.all(results.map(async result => {
+            const targetModels = await Promise.all(Object.keys(result.targets).map(async (targetID) => {
+                const targetModel = await this.openmct.objects.get(targetID);
+                const targetKeyString = this.openmct.objects.makeKeyString(targetModel.identifier);
+                const originalPathObjects = await this.openmct.objects.getOriginalPath(targetKeyString);
+
+                return {
+                    originalPath: originalPathObjects,
+                    ...targetModel
+                };
+            }));
+
+            return {
+                targetModels,
+                ...result
+            };
+        }));
+
+        return modelAddedToResults;
+    }
+
+    async searchForTags(query, abortController) {
```

Comment:
This seems like a private function as well?

Reply:
OK, just realized it's not private. Can we add some docs for this? I think tags are sufficient different from other annotation types, and generally applicable so having a public API function makes sense here.